### PR TITLE
feat: Update Karpenter pattern to use `v0.32.1` with `v1beta1` APIs

### PR DESCRIPTION
**Pull request by bryantbiggs**
# Description

- Update Karpenter pattern to use `v0.32.1` with `v1beta1` APIs
- Remove the `kubectl` provider since it is no longer maintained

### Motivation and Context

- Relates to #1675

### How was this change tested?

- [x] Yes, I have tested the PR using my local account setup (Provide any test evidence report under Additional Notes)
- [x] Yes, I have updated the [docs](https://github.com/aws-ia/terraform-aws-eks-blueprints/tree/main/docs) for this feature
- [x] Yes, I ran `pre-commit run -a` with this PR

### CODE REVIEW

**Review by rodrigobersa — COMMENTED** [review]
This have just 2 manifests, but in other examples where we have more, wouldn't be better to have a `manifests` directory?

**Comment by bryantbiggs** [comment]
> wouldn't be better to have a manifests directory?

I'm curious to know more - how and what would this directory improve?

**Comment by rodrigobersa** [comment]
> > wouldn't be better to have a manifests directory?
> 
> I'm curious to know more - how and what would this directory improve?

Just can keep the `patterns` directory more organized, separation of concerns between .tf and .yaml.

```
tree
.
├── 1.yaml
├── 2.yaml
├── 3.yaml
├── 4.yaml
├── 5.yaml
├── README.md
├── main.tf
├── outputs.tf
├── variables.tf
└── versions.tf
```

```
tree
.
├── README.md
├── main.tf
├── manifests
│   ├── 1.yaml
│   ├── 2.yaml
│   ├── 3.yaml
│   ├── 4.yaml
│   └── 5.yaml
├── outputs.tf
├── variables.tf
└── versions.tf
```

**Comment by bryantbiggs** [comment]
converting to draft until https://github.com/aws-ia/terraform-aws-eks-blueprints-addons/pull/298 is released

File: patterns/karpenter/main.tf [hunk 1]
```diff
@@ -34,28 +34,14 @@ provider "helm" {
...
-
 data "aws_ecrpublic_authorization_token" "token" {
   provider = aws.virginia
 }
 
 data "aws_availability_zones" "available" {}
 
 locals {
-  name   = basename(path.cwd)
+  name   = "ex-${basename(path.cwd)}"
```
Comment (allamand) [inline patterns/karpenter/main.tf:44]: why adding ex- ?

**Review by allamand — COMMENTED** [review]
I'm creating a cluster to see if it's all working

**Review by askulkarni2 — APPROVED** [review]
🚢

File: patterns/karpenter/README.md [hunk 2]
```diff
@@ -8,15 +8,55 @@ See [here](https://aws-ia.github.io/terraform-aws-eks-blueprints/getting-started
...
+    fargate-ip-10-0-44-106.us-west-2.compute.internal   Ready    <none>   5m12s   v1.28.2-eks-f8587cb
+    ```
+
+2. Provision the Karpenter `EC2NodeClass` and `NodePool` resources which provide Karpenter the necessary configurations to provision EC2 resources:
+
+    ```sh
+    kubectl apply -f karpenter.yaml
+    ```
+
+3. With the Karpenter resources are in place, when pods are pending in the scheduler's queue, Karpenter will provision the necessary EC2 resources to satisfy those requests. You can demonstrate this with the example deployment provided. First deploy the example deployment which has the initial number replicas set to 0:
```
Comment (bryantbiggs) [inline patterns/karpenter/README.md:29]:
```suggestion
3. Once the Karpenter resources are in place, Karpenter will provision the necessary EC2 resources to satisfy any pending pods in the scheduler's queue. You can demonstrate this with the example deployment provided. First deploy the example deployment which has the initial number replicas set to 0:
```

**Pull request merged**